src/transformers/utils/generic.py

```diff
@@ -750,3 +750,15 @@ def infer_framework(model_class):
             return "flax"
     else:
         raise TypeError(f"Could not infer framework from class {model_class}.")
+
+
+def safe_int(x):
+    import torch
+
+    return x.to(torch.int64) if torch.jit.is_tracing() else int(x)
+
+
+def safe_float(x):
+    import torch
+
+    return x.to(torch.float32) if torch.jit.is_tracing() else int(x)
```

Comment:
We need to protect the torch import here 

```suggestion
def safe_int(x):
    if not is_torch_available():
        return int(x)
        
    import torch

    return x.to(torch.int64) if torch.jit.is_tracing() else int(x)


def safe_float(x):
    if not is_torch_available():
        return int(x)

    import torch

    return x.to(torch.float32) if torch.jit.is_tracing() else int(x)
```